core/modules/parser/QSMySqlListener.cc

```diff
@@ -378,18 +395,27 @@ class RootAdapter :
         public Adapter,
         public DmlStatementCBH {
 public:
-    RootAdapter() {}
+    RootAdapter()
+    : _ctx(nullptr)
+    {}
 
     shared_ptr<query::SelectStmt>& getSelectStatement() { return _selectStatement; }
 
     void handleDmlStatement(shared_ptr<query::SelectStmt>& selectStatement) override {
         _selectStatement = selectStatement;
     }
 
-    void onExit() override {}
+    void onEnter(QSMySqlParser::RootContext * ctx) {
+        _ctx = ctx;
+    }
+
+    void onExit() override {
+        CHECK_EXECUTION_CONDITION(_selectStatement != nullptr, "Could not parse query.", _ctx);
+    }
 
 private:
     shared_ptr<query::SelectStmt> _selectStatement;
+    QSMySqlParser::RootContext * _ctx;
 };
```

Comment:
Same comment on the extra space before '*'. Otherwise it looks like a multiplication :-)